activerecord/lib/active_record/fixtures.rb

```diff
@@ -937,53 +936,38 @@ def load_fixtures
-        system(dump_cmd) or raise("Loading mysql dump failed: #{dump_cmd.inspect} resulted in an error")
-#        IO.readlines(dump_filename).join.split(";\n").each do |statement|
-#          ActiveRecord::Base.connection.execute(statement)
-#        end
-        nil
-      end
+    def ats_connect(fixture)
+      new_database_name = (fixture.to_s == 'default' ? "common" : "test_#{fixture}")
+      ActiveRecord::Base.connection.set_table_set(table_set_name: new_database_name)
+      ActiveRecord::Base.connection
+    end
 
-      # for pre_loaded_fixtures, only require the classes once. huge speed improvement
-      @@required_fixture_classes = false
+    def load_mysql_dump(dump_filename)
+      raise "Cannot be used in production!" if Rails.env == 'production'
+      config = ActiveRecord::Base.connection.config
+      dump_cmd = "mysql --user=#{Shellwords.shellescape(config[:username])} --password=#{Shellwords.shellescape(config[:password])} #{Shellwords.shellescape(config[:database])} < #{Shellwords.shellescape(dump_filename)}"
+      system(dump_cmd) or raise("Loading mysql dump failed: #{dump_cmd.inspect} resulted in an error")
+      nil
+    end
 
-      def instantiate_fixtures
-        if pre_loaded_fixtures
-          raise RuntimeError, 'Load fixtures before instantiating them.' if ActiveRecord::Fixtures.all_loaded_fixtures.empty?
-          unless @@required_fixture_classes
-            self.class.require_fixture_classes ActiveRecord::Fixtures.all_loaded_fixtures.keys
-            @@required_fixture_classes = true
-          end
-          ActiveRecord::Fixtures.instantiate_all_loaded_fixtures(self, load_instances?)
-        else
-          raise RuntimeError, 'Load fixtures before instantiating them.' if @loaded_fixtures.nil?
-          @loaded_fixtures.each_value do |fixture_set|
-            ActiveRecord::Fixtures.instantiate_fixtures(self, fixture_set, load_instances?)
-          end
+    # for pre_loaded_fixtures, only require the classes once. huge speed improvement
+    @@required_fixture_classes = false
+
+    def instantiate_fixtures
+      if pre_loaded_fixtures
+        raise RuntimeError, 'Load fixtures before instantiating them.' if ActiveRecord::Fixtures.all_loaded_fixtures.empty?
+        unless @@required_fixture_classes
+          self.class.require_fixture_classes ActiveRecord::Fixtures.all_loaded_fixtures.keys
+          @@required_fixture_classes = true
+        end
+        ActiveRecord::Fixtures.instantiate_all_loaded_fixtures(self, load_instances?)
+      else
+        raise RuntimeError, 'Load fixtures before instantiating them.' if @loaded_fixtures.nil?
+        @loaded_fixtures.each_value do |fixture_set|
+          ActiveRecord::Fixtures.instantiate_fixtures(self, fixture_set, load_instances?)
```

Comment:
Do you know the ?w=1 trick?  You can add this to the end of a pull request and it removes white-space differences.